pkg/controllers/korrel8r/controller.go

```diff
@@ -0,0 +1,129 @@
+// Copyright: This file is part of korrel8r, released under https://github.com/korrel8r/korrel8r/blob/main/LICENSE
+
+package korrel8r
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	korrel8rdeployName                  = "korrel8r"
+	OperatorNamespace      = "operators"
+)
+
+var createOrUpdateDeployment = builder.WithPredicates(predicate.Funcs{
+	UpdateFunc:  func(e event.UpdateEvent) bool { 
+		return e.ObjectNew.GetNamespace() == OperatorNamespace && e.ObjectNew.GetName() == korrel8rdeployName },
+	CreateFunc:  func(e event.CreateEvent) bool { 
+		return e.Object.GetNamespace() == OperatorNamespace && e.Object.GetName() == korrel8rdeployName },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+})
+
+// Korrel8rReconciler reconciles a Korrel8r object
+type resourceManager struct {
+	k8sClient  client.Client
+	scheme     *runtime.Scheme
+	logger     logr.Logger
+	controller controller.Controller
+	korrel8rconf Korrel8rConfiguration
+	version      string
```

Comment:
already marked by Github, but version is unused. I assume that as this is a draft it'll be filled later ;)